src/jit/importer.cpp

```diff
@@ -8654,7 +8654,12 @@ GenTree* Compiler::impFixupStructReturnType(GenTree* op, CORINFO_CLASS_HANDLE re
     // and no normalizing
     if (op->gtOper == GT_LCL_VAR)
     {
-        op->ChangeOper(GT_LCL_FLD);
+        // It is possible that we now have a lclVar of scalar type.
+        // If so, don't transform it to GT_LCL_FLD.
+        if (varTypeIsStruct(lvaTable[op->AsLclVarCommon()->gtLclNum].lvType))
```

Comment:
Could be `AsLclVar` but I guess it doesn't really matter.

Reply:
Yes, I think I'm programmed to use `AsLclVarCommon`